From 712a58665ed5ac9a4aafe80d8b88a90cf8d848d5 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:53:09 -0400
Subject: [tools] perf/machine: Add machine__get_kallsyms_filename()

Message-id: <1408953210-26343-302-git-send-email-jolsa@redhat.com>
Patchwork-id: 89022
O-Subject: [PATCH RHEL7.1 BZ1133083 301/322] perf machine: Add machine__get_kallsyms_filename()
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 15a0a8706c32bd38bff9ebf7c6ef24f32d1ea921
Author: Adrian Hunter <adrian.hunter@intel.com>
Date: Wed Jan 29 16:14:38 2014 +0200

description
===========
Separate out the logic used to make the kallsyms full path name for a
machine.  It will be reused in a subsequent patch.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/machine.c b/tools/perf/util/machine.c
index ded7459..290c2e6 100644
--- a/tools/perf/util/machine.c
+++ b/tools/perf/util/machine.c
@@ -496,19 +496,22 @@ static int symbol__in_kernel(void *arg, const char *name,
  return 1;
 }
 
+static void machine__get_kallsyms_filename(struct machine *machine, char *buf,
+        size_t bufsz)
+{
+ if (machine__is_default_guest(machine))
+  scnprintf(buf, bufsz, "%s", symbol_conf.default_guest_kallsyms);
+ else
+  scnprintf(buf, bufsz, "%s/proc/kallsyms", machine->root_dir);
+}
+
 /* Figure out the start address of kernel map from /proc/kallsyms */
 static u64 machine__get_kernel_start_addr(struct machine *machine)
 {
- const char *filename;
- char path[PATH_MAX];
+ char filename[PATH_MAX];
  struct process_args args;
 
- if (machine__is_default_guest(machine))
-  filename = (char *)symbol_conf.default_guest_kallsyms;
- else {
-  sprintf(path, "%s/proc/kallsyms", machine->root_dir);
-  filename = path;
- }
+ machine__get_kallsyms_filename(machine, filename, PATH_MAX);
 
  if (symbol__restricted_filename(filename, "/proc/kallsyms"))
   return 0;
-- 
1.7.1